cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90

```diff
@@ -105,6 +105,7 @@ module ice_comp_nuopc
   character(*), parameter      :: modName =  "(ice_comp_nuopc)"
   character(*), parameter      :: u_FILE_u = &
        __FILE__
+  real(8)                      :: timers, timere
```

Comment:
The r*8 variables should be defined as dbl_kind. 

Also, there is an existing perf_mod within the cice_wrapper.F90. For CESM, it uses GPTL but the calls are just stubs for non-CESM. I think it would better if we could implement something like that for this functionality.

Reply:
@DeniseWorthen I could not find the cice_wrapper.F90 in the cice code so I can't see the perf_mod. From the stub call interface, I am curious if the per_mod uses hash to get the actual running time for the subroutine surrounded by the "t_startf" and t_stopf at every time step or it just prints out starting time and stop time or total calling time. It would be good to understand more on how CESM is using it. 
I changed r*8 to real(dbl_kind) at this time.

Reply:
cice_wrapper_mod.F90 is in the cap directory (cicecore/drivers/nuopc/cmeps). From what I understand, CESM uses GPTL across all components and each component can access this same perf_mod so timing calls are consistent across all components. They do something similar w/ component logging, all components call into a logging routine so logging (filename, format) is consistent across components.

While this kind of across-component functionality would be really nice to have, at this point what I was suggesting is that in the cice_wrapper, the non-CESM timing calls could be defined--these calls would be stubs for CESM. That way either can compile the code but there will be no impact outside of UFS. This is also how I was planning to implement the logging function to write a small log file when the history write returns.

I've attached a copy of the perf_mod.F90 so you can see how they define the t_startf calls etc.
[perf_mod.F90.txt](https://github.com/NOAA-EMC/CICE/files/11967970/perf_mod.F90.txt)

Reply:
Thanks for the information. Do you have some CESM perf_mod output? I am not sure if this is the temporal timing we are looking for. ESMF does provide time stamp information for ESMF phases, but we have to compute the time interval from that to get the temporal timing information. From the code itself, I seem not seeing such temporal timing info, maybe I missed something.

Reply:
I don't have any perf_mod output. I have looked at the information for GPTL (https://jmrosinski.github.io/GPTL/) but I'm not an expert in this area by any means. 

In any case, at this point, I think if we can put the timing functionality into a cice_wrapper routine, that would be a workable solution. I actually think a wrapper is probably what we should do for MOM6 and WW3 also. I can try to see if I can get something to work for CICE.